tests/samples/test_api.py

```diff
@@ -188,33 +166,114 @@ async def test_find(
                 "pathoscope": False,
                 "all_read": True,
                 "labels": [label_3.id],
+                "workflows": {"aodp": "none", "nuvs": "none", "pathoscope": "none"},
             },
         ],
         session=None,
     )
+    return client
 
-    path = "/samples"
-    query = []
 
-    if find is not None:
-        query.append(f"find={find}")
+@pytest.mark.apitest
+class TestFindSamples:
+    @pytest.mark.parametrize("find", [None, "gv", "sp"])
+    async def test_term(
+        self,
+        find,
+        fake2,
+        spawn_client,
+        static_time,
+        snapshot,
+    ):
+        client = await setup_initial_test_setup(fake2, spawn_client, static_time)
+        path = "/samples"
+        query = []
+        if find is not None:
+            query.append(f"find={find}")
+        if query:
+            path += f"?{'&'.join(query)}"
+
+        resp = await client.get(path)
+        assert resp.status == 200
+        assert await resp.json() == snapshot
 
-    if per_page is not None:
-        query.append(f"per_page={per_page}")
+    @pytest.mark.parametrize("per_page,page", [(None, None), (2, 1), (2, 2)])
+    async def test_page_perpage(
+        self,
+        per_page,
+        page,
+        fake2,
+        spawn_client,
+        static_time,
+        snapshot,
+    ):
+        client = await setup_initial_test_setup(fake2, spawn_client, static_time)
+        path = "/samples"
+        query = []
+        if per_page is not None:
+            query.append(f"per_page={per_page}")
+        if page is not None:
+            query.append(f"page={page}")
+            path += f"?{'&'.join(query)}"
+
+        resp = await client.get(path)
+        assert resp.status == 200
+        assert await resp.json() == snapshot
 
-    if page is not None:
-        query.append(f"page={page}")
+    @pytest.mark.parametrize("labels", [None, [3], [2, 3], [0]])
+    async def test_labels(
+        self,
+        labels,
+        fake2,
+        spawn_client,
+        static_time,
+        snapshot,
+    ):
+        client = await setup_initial_test_setup(fake2, spawn_client, static_time)
+        path = "/samples"
+        query = []
 
-    if labels is not None:
-        label_query = "&label=".join(str(label) for label in labels)
-        query.append(f"label={label_query}")
+        if labels is not None:
+            label_query = "&label=".join(str(label) for label in labels)
```

Comment:
Looks like your appending "&" both here and when you append the query to the path